net.bobbo.player-controller/player_scripts/crouch.gd

```diff
@@ -14,30 +14,60 @@ var crouch_current: float = 0
 var crouch_target: float = 0
 const crouch_action: String = "player_crouch"
 
-## default_player_height is needed for proper uncrouching height calculations 
+## _default_player_height is needed for proper uncrouching height calculations 
 ## if the maximum player height were to change during gameplay at any point this 
 ## value would need to somehow be updated to account for that default is 2 
 ## because the default height in player_controller is 2
 var _default_player_height: float = 2
-var _should_be_crouched: bool = false
+
+var _ray_cast: RayCast3D = null
 
 const _standing_height_multiplier: float = 1
 const _crouching_height_multiplier: float = 0.5
 
 #
 #	Functions
-#
+# 
 
 func player_ready():
 	_default_player_height = player.height
+	
+	_ray_cast = RayCast3D.new()
+	_ray_cast.add_exception(player)
+	
+	# I split this up into a variable to keep it behind the second line
+	var inital_target_position = Vector3(0, _default_player_height - player.position.y, 0)
+	_ray_cast.target_position = player.up_direction * inital_target_position
+	add_child(_ray_cast)
```

Comment:
Player behavior scripts extend from Node, which means that they don't have a position. Adding the RayCast3D as a child will sorta put it in this limbo where it doesn't follow the player, because it's direct parent doesn't have a position.

You may want to consider parenting `_ray_cast` to `player` instead

Reply:
Parenting it to player would allow for the raycast to follow the player's position and also make the ray automatically exclude it so this is definitely something I'm implementing